tests/python/test_with_sklearn.py

```diff
@@ -12,7 +12,7 @@
 
 rng = np.random.RandomState(1994)
 
-pytestmark = pytest.mark.skipif(**tm.no_sklearn())
+pytestmark = [pytest.mark.skipif(**tm.no_sklearn()), pytest.mark.timeout(30)]
```

Comment:
Which test might timeout?

Reply:
Times taken from CI:

```
36.49s call     python/test_with_sklearn.py::test_parameter_tuning
34.58s call     python/test_with_sklearn.py::test_RFECV
```

I made the test under 30s now, but I am placing timeouts to prevent others from adding any slower tests than currently exist.

Reply:
Ideally we will be targeting <5s for python tests with some few exceptions.

Reply:
Got it, thank you for reducing the time!